### ISSUE TITLE
Custom BackPropagation & Neural Network architecture for a Regression problem

### ISSUE THREAD

**Issue by diliprk** [MichalDanielDobrzanski/DeepLearningPython35#22]
Hi Michal, 
 Thank you for putting up Nielsen's code in Python 3.5. I am trying to build a backpropagation algorithm for a custom physics guided loss function where a neural network works in tandem with a physics-based simulator for a regression problem. Since heavy customizations are required, libraries like Keras are not suitable for the job and I have to build the neural network from scratch. I read the first two chapters of Nielsen's book and found the explanation and code quite suitable to use as a starting template. My clarifications are:
1.) Can I start with `network.py` or should I use the improved versions (`network2.py` & `network3.py` as a starting base? 
2.) Which activation function should I use? Can I continue using `Sigmoid` as in the code or should I implement `ReLu `or another activation function for regression predictions?
3.) What parts of the code I should change for this regression problem so that the network architecture accepts an `np.array` of 7 input features and 1 output column, because right now it does not accept multidimensional array in the `training data`?

**Comment by MichalDanielDobrzanski** [comment]
Hey @diliprk. It is up to you to find the best neural network architecture. This is all about it. You must validate your architectures against your data set.

I am closing this issue.
